pkg/event/processor_test.go

```diff
@@ -449,3 +449,39 @@ func TestChanQueueEventProcessor_ProcessBatch(t *testing.T) {
 		assert.True(t, len(logEvent.Event.Visitors) >= 1)
 	}
 }
+
+func BenchmarkProcessor(b *testing.B) {
+	exeCtx := utils.NewCancelableExecutionCtx()
+	processor := NewBatchEventProcessor(
+		WithQueueSize(1000),
+		WithQueue(NewInMemoryQueue(1000)),
+		WithFlushInterval(10 * time.Millisecond),
+		WithEventDispatcher(NewMockDispatcher(100, false)))
+	processor.Start(exeCtx)
+
+	impression := BuildTestImpressionEvent()
+	conversion := BuildTestConversionEvent()
+
+	for i := 0; i < b.N; i++ {
+		processor.ProcessEvent(impression)
+		processor.ProcessEvent(conversion)
+		time.Sleep(1 * time.Millisecond)
```

Comment:
Doesn't adding the `time.Sleep` nullify the benchmark?

Reply:
There is literally no other way to let the go routines run that put in a sleep.  The sleep is now 500 nanoseconds.